homeassistant/components/sensor/zamg.py

```diff
@@ -140,40 +145,90 @@ def __init__(self, logger, station_id):
         self._station_id = station_id
         self.data = {}
 
+    @property
+    def last_update(self):
+        """Return the timestamp of the most recent data."""
+        date, time = self.data.get('update_date'), self.data.get('update_time')
+        if date is not None and time is not None:
+            return datetime.strptime(date + time, '%d-%m-%Y%H:%M').replace(
+                tzinfo=pytz.timezone('Europe/Vienna'))
```

Comment:
Using a `datetime` object makes it much easier to calculate whether new data is available in the `update` method, so I'd rather leave it.